certbot-apache/certbot_apache/dualparser.py

```diff
@@ -0,0 +1,266 @@
+""" Tests for ParserNode interface """
+from certbot_apache import assertions
+from certbot_apache import augeasparser
+
+
+class DualNodeBase(assertions.InterfaceAssertions):
```

Comment:
I think we should have a place to specify the classes that should be used for both primary and secondary inside these objects. This allows us to easily plug in the apacheconfig implementation and swap which is the primary without having to modify every place the class is referenced in the config.